### DISCUSSION TITLE
Week 8 Assignment: code + office hours

### DISCUSSION THREAD

**Discussion by hamzamajoka** [gui2de/ppol768-spring23#274] [Announcements]
@gui2de/ppol768-spring23 

I've uploaded the do files from yesterday's class (github) + class recording (box). 

I'll be holding office hours today (Friday) 4-5 pm in the gui2de office in case you have any questions regarding the week 8 assignment.

**Comment by genjuro86** [comment]
Dear Ali,
Good morning.

While I was doing week 8 assignment, I got an error message saying _"observation number out of range
    Observation number must be between 10,000 and 2,147,483,619.  (Ob_servation numbers are typed without
    commas.)"_. 

And my codes are as below:

* Part 1: Sampling noise in a fixed population (1~2)

clear

set seed 1234
set obs 10000

gen e1 = rnormal()
gen e2 = runiform()
gen random_num = rnormal()

gen x=random_num+e1
gen y=5+1*x+(e1+e2)

save Part1_data

* Part 1: Sampling noise in a fixed population (3)
clear

capture program drop normal_reg 
program define normal_reg, rclass 
	use "C:\Genjuro\Georgetown_University\2_Spring_2023\Research_Design_and_Implementation\Week 8\Part1_data.dta"
	syntax, samplesize(integer)
	set obs `samplesize'
	reg y x
end

normal_reg, samplesize(10)


=> In this situation, could you please help me out why the code "syntax, samplesize(integar) - normal_reg, samplesize(10)" doesn't work properly?

Thank you in advance,

**Reply by hamzamajoka** [reply]
I think there are a couple of issues:

1) the syntax part should be right after the program define part:

```
capture program drop normal_reg
program define normal_reg, rclass
syntax, samplesize(integer)

```

2) set obs command doesn't make a lot of sense when you are loading the dataset in the previous line. Maybe write down in simple bullets what each line of code is trying to do. That is very helpful when trying to debug your code.

**Comment by marbruno** [comment]
Will there be office hours tomorrow?

**Reply by bbdaniels** [reply]
I will be available before and after class, yes!

**Reply by hamzamajoka** [reply]
I'll have office hours today (Monday) in the gui2de office from 3 pm - 4 pm.
